Sources/TokamakCore/Views/Toolbar.swift

```diff
@@ -0,0 +1,127 @@
+  public static let principal: ToolbarItemPlacement = .init(rawValue: 1 << 1)
+  public static let navigation: ToolbarItemPlacement = .init(rawValue: 1 << 2)
+  public static let primaryAction: ToolbarItemPlacement = .init(rawValue: 1 << 3)
+  public static let status: ToolbarItemPlacement = .init(rawValue: 1 << 4)
+  public static let confirmationAction: ToolbarItemPlacement = .init(rawValue: 1 << 5)
+  public static let cancellationAction: ToolbarItemPlacement = .init(rawValue: 1 << 6)
+  public static let destructiveAction: ToolbarItemPlacement = .init(rawValue: 1 << 7)
+  public static let navigationBarLeading: ToolbarItemPlacement = .init(rawValue: 1 << 8)
+  public static let navigationBarTrailing: ToolbarItemPlacement = .init(rawValue: 1 << 9)
+  public static let bottomBar: ToolbarItemPlacement = .init(rawValue: 1 << 10)
+}
+
+public protocol AnyToolbarItem {
+  var placement: ToolbarItemPlacement { get }
+  var anyContent: AnyView { get }
+}
+
+public struct ToolbarItem<ID, Content>: View, AnyToolbarItem where Content: View {
+  public let id: ID
+  public let placement: ToolbarItemPlacement
+  let showsByDefault: Bool
+  let content: Content
+  public var anyContent: AnyView { AnyView(content) }
+  public init(id: ID,
+              placement: ToolbarItemPlacement = .automatic,
+              showsByDefault: Bool = true,
+              @ViewBuilder content: () -> Content) {
+    self.id = id
+    self.placement = placement
+    self.showsByDefault = showsByDefault
+    self.content = content()
+  }
+
+  public var body: Content {
+    content
+  }
+}
+
+extension ToolbarItem where ID == () {
+  public init(placement: ToolbarItemPlacement = .automatic,
+              @ViewBuilder content: () -> Content) {
+    self.init(id: (), placement: placement, showsByDefault: true, content: content)
+  }
+}
+
+extension ToolbarItem: Identifiable where ID: Hashable {}
+
+/// This is a helper class that works around absence of "package private" access control in Swift
+public struct _ToolbarItemProxy<ID, Content> where Content: View {
+  public let subject: ToolbarItem<ID, Content>
+
+  public init(_ subject: ToolbarItem<ID, Content>) { self.subject = subject }
+
+  public var placement: ToolbarItemPlacement { subject.placement }
+  public var showsByDefault: Bool { subject.showsByDefault }
+  public var content: Content { subject.content }
+}
+
+public protocol _AnyToolbarContainer {
+  var anyContent: AnyView { get }
+}
+
+public struct _ToolbarContainer<ID, Content, Wrapped>: View where Wrapped: View {
+  public let id: ID
+  public let content: Content
+  public let child: Wrapped
+  public var body: Wrapped {
+    child
+  }
+}
+
+extension _ToolbarContainer: _AnyToolbarContainer where Content: View {
+  public var anyContent: AnyView { AnyView(content) }
+}
+
+extension View {
+  @_disfavoredOverload public func toolbar<Content>(@ViewBuilder content: () -> Content) -> some View where Content: View {
```

Comment:
<!--
  1 failure:  Line should be 10...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: Sources/TokamakCore/Views/Toolbar.swift;
  Line: 115;
-->
- :no_entry_sign: Line should be 100 characters or less: currently 123 characters (`line_length`)